caddyconfig/caddyfile/parse.go

```diff
@@ -214,6 +214,11 @@ func (p *parser) addresses() error {
 			break
 		}
 
+		// Users commonly forget to place a space between the address and the '{'
+		if strings.HasSuffix(tkn, "{") {
+			return p.Errf("Addresses cannot end in '{', got '%s' - did you mean to put a space between the address and '{'?", tkn)
```

Comment:
`caddy fmt` does fix it, but like only 1% of users actually run `caddy fmt` lol, and they don't know to when they get this problem cause the error message they would get with this problem is misleading (says `directive "example.com{" doesn't exist` or something useless like that, or even actually runs but then tries to do ACME but fails cause a domain with `{` doesn't qualify)

:+1: to the comment -- I sorta feel it should mention the brace is `{` but :man_shrugging: hopefully users understand that that's a brace.

Why not the word "cannot"? I feel like this would be better?

```suggestion
			return p.Errf("Site addresses cannot end with a curly brace: '%s' - put a space between the token and the brace", tkn)
```

Reply:
Technically, block keys *can* end a curly brace (but that would be weird). The only reason site addresses don't is because of hostname and URL encodings.